.github/workflows/make-version.yml

```diff
@@ -21,6 +21,7 @@ jobs:
         uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633  # v4.1.2
         with:
           ref: ${{ github.ref }}
+          fetch-depth: 0
```

Comment:
Could we add a comment above to explain what is this for, and what implications it brings? 

I just want to avoid us thinking "we don't need this" in 6 months and silently break the versioning process.